### PR TITLE
refactor: make the concept of storage clients abstract

### LINKED ISSUE #40: Abstract storage-local and client, use client based on `Actor.start()` call
Currently this works based on `Apify.isAtHome()`, but we want the apify-client integration only on apify platform, where users need `Actor.start()` anyway. So we can make this cleaner, `Actor.start()` will change the default storage implementation on the singleton config instance, rest will be transparent.

Storage layer should be defined via simple interface that will match both storage-local and apify-client. Users should be able to provide their own implementation.

### PR DESCRIPTION
**Pull request by B4nan**
TODO:
- [x] https://github.com/apify/apify-storage-local-js/issues/42

Closes #40

### CODE REVIEW

**Review by mnmkng — COMMENTED** [review]
Left just some nitpicks and comments we should not forget about.

File: packages/apify/src/platform_event_manager.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * [EventEmitter](https://nodejs.org/api/events.html#events_class_eventemitter)
+ * class that emits various events from the SDK or the Apify platform.
+ * The event emitter is initialized by calling the {@link Actor.main} function.
+ *
+ * **Example usage:**
+ *
+ * ```javascript
+ * Actor.events.on('cpuInfo', (data) => {
+ *   if (data.isCpuOverloaded) console.log('Oh no, the CPU is overloaded!');
+ * });
```
[the remark this answers is not in the document]
Comment (B4nan) [reply packages/apify/src/platform_event_manager.ts:17]: the docs are outdated, its actually `Actor.eventManager.on()` but will add a shortcut for `Actor.on` too

(we can rename the property to `events`, I don't mind, but given `events` was an `EventEmitter` before and now its `EventManager` instance (that holds the emitter inside), I used a different name)

File: packages/apify/src/platform_event_manager.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ */
+export class PlatformEventManager extends EventManager {
+    /** Websocket connection to actor events. */
+    private eventsWs?: WebSocket;
+
+    /**
+     * Initializes `Actor.events` event emitter by creating a connection to a websocket that provides them.
+     * This is an internal function that is automatically called by `Actor.main()`.
+     */
+    override async start() {
```
[the remark this answers is not in the document]
Comment (B4nan) [reply packages/apify/src/platform_event_manager.ts:51]: let's have `init` and `close` instead. i'd say `init` is better than `connect`, as it does not connect to anything locally (only the platform one), and it does more than just connecting.

File: packages/apify/src/platform_event_manager.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ *   Note that the `persistState` event is provided merely for user convenience,
+ *   you can achieve the same effect using `setInterval()` and listening for the `migrating` event.
+ */
+export class PlatformEventManager extends EventManager {
+    /** Websocket connection to actor events. */
+    private eventsWs?: WebSocket;
+
+    /**
+     * Initializes `Actor.events` event emitter by creating a connection to a websocket that provides them.
+     * This is an internal function that is automatically called by `Actor.main()`.
```
[the remark this answers is not in the document]
Comment (B4nan) [reply packages/apify/src/platform_event_manager.ts:49]: ok `Actor.init` it is, why not :]

**Pull request merged**